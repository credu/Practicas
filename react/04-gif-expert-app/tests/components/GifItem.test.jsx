import { fireEvent, render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Pruebas para Giftitem', () => {
    const title = "saitama";
    const url = "https://www.google.com/";

    test('Debe hacer match con el snapshot', () => {
        const { container } = render( <GifItem title={ title } url= { url } /> );
        expect( container ).toMatchSnapshot();
    });

    test('Debe de mostrar la imagen con el url y el alt indicado', () => {
        render( <GifItem title={ title } url={ url } /> );
        const { src, alt } = screen.getByRole("img")
        expect( src ).toBe( url );
        expect( alt ).toBe( title );
    })

    test('Debe de mostrar el titulo en un componente', () => {
        render( <GifItem title={ title } url={ url } /> );
        expect(screen.getByText(title)).toBeTruthy();
    })
})