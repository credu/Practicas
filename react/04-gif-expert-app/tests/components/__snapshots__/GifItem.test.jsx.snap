// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para Giftitem Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="saitama"
      src="https://www.google.com/"
    />
    <p>
      saitama
    </p>
  </div>
</div>
`;
